core/startup/config_file_watcher.go

```diff
@@ -73,11 +73,13 @@ func WatchConfigDirectory(configDir string, appConfig *config.ApplicationConfig)
 				if !ok {
 					return
 				}
-				if event.Has(fsnotify.Write) {
+				if event.Has(fsnotify.Write) || event.Has(fsnotify.Create) || event.Has(fsnotify.Rename) {

```

Comment:
you can condense this down to `if event.Has(fsnotify.Write | fsnotify.Create | fsnotify.Rename) {`